src/platform/src/main/java/org/geoserver/platform/GeoServerExtensionsWithoutFiltering.java

```diff
@@ -0,0 +1,139 @@
+/* (c) 2022 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.platform;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import org.geotools.util.factory.FactoryRegistry;
+import org.springframework.context.ApplicationContext;
+
+/** Implementation of GeoServerExtensions that bypasses Exclusion Filters for UI purposes */
+public class GeoServerExtensionsWithoutFiltering extends GeoServerExtensions {
```

Comment:
Ouch, this one looks bad, I see why you have created one, but wondering if it's necessary.

At the beginning I've suggested to use an exclusion filter myself, but in the end both GetFeature and GetCapabilities have their own local checks, so it's a bit of "belt and suspenders" that's adding extra complication.

It seems that if you removed the bean filter and this class, the functionality would still be there. Am I wrong?

Reply:
done